### PR TITLE
Added the ability to configure by profile aws

### PR DESCRIPTION
**Pull request by EduardoMartinezCatala**

### CODE REVIEW

File: lib/.env.example [hunk 1]
```diff
@@ -1,6 +1,7 @@
 AWS_ENVIRONMENT=development
 AWS_ACCESS_KEY_ID=your_key
 AWS_SECRET_ACCESS_KEY=your_secret
+AWS_PROFILE=NO_NEED_KEY
```
Comment (DeviaVir) [inline lib/.env.example]: Perhaps better to leave it empty for this example

File: lib/main.js [hunk 2]
```diff
@@ -480,12 +482,19 @@ Lambda.prototype.deploy = function (program) {
...
-        accessKeyId: program.accessKey,
-        secretAccessKey: program.secretKey,
         region: region
       };
 
-      if (program.sessionToken){
+      if (program.profile) {
+        aws.config.credentials = new aws.SharedIniFileCredentials({
+                     profile: program.profile
+                 });
```
Comment (DeviaVir) [inline lib/main.js]: seems like a strange way to indent this?

Comment (EduardoMartinezCatala) [inline lib/main.js]: sorry, the editor i use Automatically indent the file that way

**Comment by DeviaVir** [comment]
I'm superseding this PR by https://github.com/motdotla/node-lambda/pull/144 which does not have all of the indent/code changes generated by an editor, but does include all the changes.

Thanks @EduardoMartinezCatala for your contribution! It should land in [0.8.8](https://github.com/motdotla/node-lambda/issues/139)

**Pull request closed without merging**